labelme/widgets/canvas.py

```diff
@@ -383,16 +411,40 @@ def mousePressEvent(self, ev):
                         self.line[0] = self.current[-1]
                         if int(ev.modifiers()) == QtCore.Qt.ControlModifier:
                             self.finalise()
+                    elif self.createMode == "ai_polygon":
```

Comment:
I think this is a bug, which will make the last point the point of the positive sample. It should be modified as follows.

`                        if ev.modifiers() & QtCore.Qt.ControlModifier:
                            self.finalise()
                        else:
                            self.current.addPoint(
                                self.line.points[1],
                                label=self.line.point_labels[1],
                            )
                            self.line.points[0] = self.current.points[-1]
                            self.line.point_labels[0] = self.current.point_labels[
                                -1
                            ]`

Reply:
If you use `Ctrl/Cmd+Shift+Click`, it will be negative sample to close the polygon, so it should be okay?

Reply:
If I don't use Ctrl+leftClick in the target, I will add a negative sample point as a positive sample point to 'self. current' and end up with an error. We don't know where the user will click